File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    hp = health
    time, recovery, bonus = bandage
    start = 1
    
    for t, attack in attacks:
        # 공격받기 전까지의 hp계산
        hp += ((t - start) // time) * bonus + (t - start) * recovery
        # 공격받은 후로 start 초기화
        start = t + 1
        # 최대 회복량 설정
        if hp >= health: hp = health
        # 체력에서 피해량만큼 감소
        hp -= attack
        if hp <= 0: return -1
    
    return hp

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.js
function solution(bandage, health, attacks) {
    const [time, recovery, bonus] = bandage;
    let start = 1;
    let hp = health;
    
    for (let [t, attack] of attacks){
        hp += (Math.trunc((t - start)/time) * bonus) + ((t - start) * recovery);
        start = t + 1
        if (hp >= health) hp = health;
        hp -= attack;
        if (hp <= 0) return -1
    }
    
    return hp;
}

